src/rgw/rgw_rest_s3.h

```diff
@@ -678,42 +719,65 @@ class RGWLDAPAuthEngine: RGWTokenBasedAuthEngine
   RGWAuthApplier::aplptr_t authenticate() const override;
 };
 
-class RGWLDAPTokenExtractor : public RGWTokenBasedAuthEngine::Extractor {
+class RGWS3V2LocalAuthEngine: RGWS3V2AuthEngine
+{
 protected:
-  const req_state * const s;
+  req_state* const s;
+  RGWRados* const store;
+  const RGWLocalAuthApplier::Factory* const apl_factory;
 
 public:
-  RGWLDAPTokenExtractor(const req_state * const s)
-    : s(s) {
+  RGWS3V2LocalAuthEngine(req_state* const s,
+                          RGWRados* const store,
+                          const Extractor& extr,
+                          const RGWLocalAuthApplier::Factory * const apl_factory)
+    : RGWS3V2AuthEngine(s->cct, extr),
+      s(s),
+      store(store),
+      apl_factory(apl_factory) {
+  }
+
+  const char* get_name() const noexcept override {
+    return "RGWS3V2LocalAuthEngine";
   }
-  std::string get_token() const override;
+  bool is_applicable() const noexcept override;
+  RGWAuthApplier::aplptr_t authenticate() const override;
 };
 
-class RGWGetPolicyLDAPTokenExtractor : public RGWTokenBasedAuthEngine::Extractor {
+class RGWGetPolicyTokenExtractor:public RGWS3V2AuthEngine::Extractor {
+private:
   std::string access_key_id;
+  std::string signature;
+
 public:
-  RGWGetPolicyLDAPTokenExtractor(std::string access_key_id) {
-    access_key_id = std::move(access_key_id);
+  RGWGetPolicyTokenExtractor(std::string access_key_id, std::string signature) {
+    access_key_id = std::move(access_key_id),
+    signature = std::move(signature);
   }
 
-  std::string get_token() const {
-    return access_key_id;
+  void get_auth_keys(std::string& access_key_id,
```

Comment:
Not initializing 'qsr' in this piece of code won't make a difference, or cause any harm. Since the authentication engine (ldap auth) is not using this parameter now.